docs/ref/migration-operations.txt

```diff
@@ -423,9 +423,17 @@ if ``atomic=True`` is passed to the ``RunPython`` operation.
 A highly specialized operation that let you mix and match the database
 (schema-changing) and state (autodetector-powering) aspects of operations.
 
-It accepts two lists of operations, and when asked to apply state will use the
-state list, and when asked to apply changes to the database will use the database
-list. Do not use this operation unless you're very sure you know what you're doing.
+It accepts two lists of operations. When asked to apply state, it will use
+the ``state_operations`` list. When asked to apply changes to the database, it
+will use the ``database_operations`` list.
+
+This is a generalized version of :class:`RunSQL`'s ``state_operations``
+argument. It's worth exercising caution and checking your state and database
+operations carefully with :djadmin:`makemigrations` and :djadmin:`sqlmigrate`
+respectively.
```

Comment:
I think a more clear warning would be good here, especially for less experienced users. Can we maybe prepend this paragraph with something like "This operation can cause severe problems if not used carefully."?

Reply:
I think the sentence "It's worth exercising caution and checking ..." is a good enough warning, as it also explains some tools for checking. I don't like vague warnings about things going wrong without any details as it can be read as condescending, and users still need to experiment to understand how things work.

Reply:
But if you have some proposed wording, I'm all ears (...or eyes).

Reply:
I'm not sure if the current wording would make the dangers clear enough to me if I was a beginner, to be honest. People often read the docs in a hurry. Many are also non-native speakers.

Personally I liked the clarity of "do not use this unless you know what you're doing". But if it sounds condescending to people, let's change it, yup. I'm just wondering if there is a formulation that is not condescending, but equally clear.

Maybe something like "If the actual state of the database and Django's view of the state get out of sync, this can break the migration framework. Therefore, it's worth exercising caution [...]"?

Overall it is not a major issue though, I just wanted to share my reservations :)

Reply:
I've expanded the paragraph to include the warning about breaking the migration framework - that's great wording, thanks!